code/modules/mob/living/carbon/xenomorph/castes/shrike/shrike.dm

```diff
@@ -0,0 +1,140 @@
+/mob/living/carbon/xenomorph/shrike
+	caste_base_type = /mob/living/carbon/xenomorph/shrike
+	name = "Shrike"
+	desc = "A large, lanky alien creature. It seems psychically unstable."
+	icon = 'icons/Xeno/2x2_Xenos.dmi'
+	icon_state = "Shrike Walking"
+	attacktext = "bites"
+	attack_sound = null
+	friendly = "nuzzles"
+	wall_smash = FALSE
+	health = 240
+	maxHealth = 240
+	plasma_stored = 300
+	speed = -0.2
+	pixel_x = -16
+	old_x = -16
+	drag_delay = 3 //pulling a medium dead xeno is hard
+	tier = XENO_TIER_FOUR
+	upgrade = XENO_UPGRADE_ZERO
+	xeno_explosion_resistance = 2 //some resistance against explosion stuns.
+	job = ROLE_XENO_QUEEN
+	var/shrike_flags = SHRIKE_FLAG_PAIN_HUD_ON
+	var/mob/living/carbon/human/psychic_victim
+
+	actions = list(
+		/datum/action/xeno_action/xeno_resting,
+		/datum/action/xeno_action/regurgitate,
+		/datum/action/xeno_action/plant_weeds,
+		/datum/action/xeno_action/lay_egg,
+		/datum/action/xeno_action/call_of_the_burrowed,
+		/datum/action/xeno_action/choose_resin,
+		/datum/action/xeno_action/activable/secrete_resin,
+		/datum/action/xeno_action/activable/corrosive_acid,
+		/datum/action/xeno_action/psychic_whisper,
+		/datum/action/xeno_action/activable/psychic_fling,
+		/datum/action/xeno_action/activable/psychic_choke,
+		/datum/action/xeno_action/toggle_pheromones
+		)
+	inherent_verbs = list(
+		/mob/living/carbon/xenomorph/proc/vent_crawl
+		)
+
+// ***************************************
+// *********** Init
+// ***************************************
+/mob/living/carbon/xenomorph/shrike/Initialize()
+	RegisterSignal(src, COMSIG_HIVE_XENO_MOTHER_CHECK, .proc/is_burrowed_larva_host)
```

Comment:
move this to the call of the burrowed action datum, register on owner with a proc on the datum.